java/server/src/org/openqa/grid/web/servlet/api/v1/Proxy.java

```diff
@@ -39,7 +41,11 @@
     }
 
     final String proxyToFind = query.replaceAll("^/", "");
-    final RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
+    RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
```

Comment:
@mach6 

> The problem is a proxy id can be anything a user wants it to be. Only by default or if the (user followed the default convention) will it be http://{host}:{port}

In that case the earlier change itself was sufficient, because it was basically querying based on whatever the user passes. So we can revert it back. 

> So, we really need to one way hash the proxy id or similar and require the user to match all proxy id's to their hash by making a GET on a /proxies resource (or I guess it could be the /nodes resource)..

This additional step IMHO Its not user friendly. I don't think there is anything wrong with using whatever the user has provided as the `id` for a proxy (even if it means that we are going to default to the url of the remoteHost) which should also be fine IMO rather than resorting to hashing it and making it more complex.

> If you look at java -jar selenium-server-standalone -role node -h and the documentation for id it says this;


I don't remember seeing this in Selenium `2.53.1`.  Following is what I see

```
java -jar selenium-server-standalone-2.53.1.jar -role node -help

-------------------------------
Running as a grid node:
-------------------------------

Usage: java -jar selenium-server.jar -role node [options]

  -host:
    <IP | hostname> : usually not needed and determined
      automatically. For exotic network configuration, network with
      VPN, specifying the host might be necessary.

  -port:
    <xxxx> : the port the remote/hub will listen on. Default to 4444.

  -cleanupCycle:
    <XXXX> in ms. How often a proxy will check for timed out thread.

  -timeout:
    <XXXX>  the timeout in seconds before the hub automatically ends
      a test that hasn't had any activity in the last X seconds. The
      browser will be released for another test to use. This typically
      takes care of the client crashes.

  -browserTimeout:
    The timeout in seconds a browser can hang

  -hub:
    <http://localhost:4444/grid/register> : the url that will be used
      to post the registration request. This option takes precedence
      over -hubHost and -hubPort options.

  -hubHost:
    <IP | hostname> : the host address of a hub the registration
      request should be sent to. Default to localhost. Option -hub
      takes precedence over this option.

  -hubPort:
    <xxxx> : the port listened by a hub the registration request
      should be sent to. Default to 4444. Option -hub takes precedence
      over this option.

  -proxy:
    the class that will be used to represent the node. By default
      org.openqa.grid.selenium.proxy.DefaultRemoteProxy.

  -maxSession:
    max number of tests that can run at the same time on the node,
      independently of the browser used.

  -registerCycle:
    how often in ms the node will try to register itself again.Allow
      to restart the hub without having to restart the nodes.

  -nodePolling:
    in ms. Interval between alive checks of node how often the hub
      checks if the node is still alive.

  -unregisterIfStillDownAfter:
    in ms. If the node remains down for more than
      unregisterIfStillDownAfter millisec, it will disappear from the
      hub.Default is 1min.

  -downPollingLimit:
    node is marked as down after downPollingLimit alive checks.

  -nodeStatusCheckTimeout:
    in ms. Connection and socket timeout which is used for node alive
      check.

This synopsis lists options available in node role only. To get help
  on the command line options available for other roles run the server
  with -help name and the corresponding -role name value.

```
Just thinking aloud here. I am not sure if this is something that should be left to the user to customize because this is internally used by the node to uniquely identify itself (which you called out). In that case, should a user really fiddle around with this ? Its not altering or tweaking the behavior of the node in any way (which I believe is what the different parameters in the `GridNodeConfiguration` stand for. So all that said and done, is this something that we should really worry about ? Thoughts ?